lib/preview/video.js

```diff
@@ -68,34 +70,58 @@ var process = module.exports.process = function(ctx, callback) {
 };
 
 /**
- * Extract the first frame as a PNG image
+ * Extract the (approximately) middle frame as a PNG image
  *
  * @param  {Context}    ctx                          The job context
  * @param  {Function}   callback                     Standard callback function
  * @param  {Object}     callback.err                 An error object, if any
  * @param  {String}     callback.imagePath           The path to the extracted image
  */
-var extractFirstFrame = function(ctx, callback) {
-  var extractCommand = [
-    'ffmpeg',
-    util.format('-i "%s"', ctx.sourcePath),
-    '-vframes 1',
-    path.join(ctx.directory, 'frame%d.png')
+var extractMiddleFrame = function(ctx, callback) {
+  var getDurationCommand = [
+    'ffprobe',
+    '-v error',
+    '-select_streams v:0',
+    '-show_entries stream=duration',
+    '-of default=noprint_wrappers=1:nokey=1',
+    util.format('"%s"', ctx.sourcePath)
   ].join(' ');
 
-  exec(extractCommand, {'timeout': 200000}, function(err, stdout, stderr) {
+  exec(getDurationCommand, {'timeout': 30000}, function(err, stdout, stderr) {
+    var timecode = 0;
     if (err) {
-      var msg = util.format('Error while extracting first frame image from video: %s', err.message);
-      return callback({'code': 500, 'msg': msg});
+      log.error({'err': err, 'path': ctx.sourcePath}, 'Error getting video duration, will attempt to use first frame');
+    } else {
+      var duration = Number(stdout);
+      if (Number.isNaN(duration)) {
```

Comment:
@paulkerschen Forgive my ignorance: will the above `isNaN` behave as expected?  See https://ariya.io/2014/05/the-curious-case-of-javascript-nan  (e.g., `Number.isNaN('hello');  // false`?!)  Am I misreading?